src/main/java/org/jenkinsci/plugins/mesos/JenkinsScheduler.java

```diff
@@ -583,7 +579,7 @@ private void getContainerInfoBuilder(Offer offer, Request request, String slaveN
                           .setContainerPort(portMapping.getContainerPort()) //
                           .setProtocol(portMapping.getProtocol());
 
-                  int portToUse = portMapping.getHostPort() == null ? portsToUse.get(portToUseIndex++) : portMapping.getHostPort();
+                  int portToUse = portMapping.getHostPort() == null ? iterator.next() : portMapping.getHostPort();
```

Comment:
May throw runtime exception. Not a bug, because the original code behaves in such way (BTW Exception classes are different)